CONTRIBUTING.md

```diff
@@ -54,45 +54,75 @@ An essential part of getting your change through is to make sure all existing te
 * Make sure you are at the directory where pom.xml is located.
 
 ##### Unit Tests
-* Run the following:
-
-```
-mvn test
-```
+* Run the following: 
+    ```
+    mvn test
+    ```
 
 ##### Integration Tests
-* By default, the integration tests are not executed. In case you are interested in executing 
-  them, disable the `skipITs` property.
-* The following environment variables are required to run the integration tests. 5, 6, and 7 are
-  only required when running a windows integration test.
-
-1. GOOGLE_PROJECT_ID
-1. GOOGLE_CREDENTIALS
-1. GOOGLE_REGION
-1. GOOGLE_ZONE
-1. GOOGLE_BOOT_DISK_PROJECT_ID
-1. GOOGLE_BOOT_DISK_IMAGE_NAME
-1. GOOGLE_JENKINS_PASSWORD
-1. GOOGLE_SA_NAME
-
-* Run the following:
-```
-mvn verify -DskipITs=false
-```
+
+Integration tests actually provision instances in a GCP project, run pipeline, take snapshot etc.
+Therefore, they are disabled in the CI and expected to be executed by contributors in their laptop itself.
+
+Reasons for disabling integration test in CI,
+* getting provisioning GCP infra is not possible
+* even if we did get a GCP infra setup in the CI, it is risky to expose that, as someone can abuse the CI.
+
+By default, integration tests are skipped from the maven goals, need to enable using the `skipITs` property.
+
+Steps to execute integration test
+* Prepare VM images  
+  (ideally we should automate this see idea [here](https://github.com/jenkinsci/google-compute-engine-plugin/pull/492#discussion_r1892705637))  
+  The jenkins agent images need to have java installed. We have a packer script to create the image and upload to your configured GCP project.
+  The scripts are located in [testimages/linux](./testimages/linux)
+  Navigate to the directory and execute,
+  ```bash
+  bash setup-gce-image.sh
+  ```
+  If you want to execute the integration tests for `non-standard-java` location, then create non-standard-java image as,
+  ```bash
+  bash setup-gce-image.sh non-standard-java
+  ```
+  If you want to delete the images or recreate them, use the arguments `--recreate` or `--delete`.
+
+* Create a service account with relevant access - See [Refer to IAM Credentials](Home.md#iam-credentials)   
+
+* Export these mandatory environment variable   
+  ```bash
+  export GOOGLE_PROJECT_ID=your-project-id
+  export GOOGLE_CREDENTIALS=/path/to/sa-key.json
```

Comment:
Oh, this is bad. Why can the tests not use your default credentials chain? You should generally not be creating a SA private key for GCP.

Reply:
https://github.com/jenkinsci/google-compute-engine-plugin/blob/79b8ffdb9a7ad456b7ea013e88802f2ea3cd1046/docs/Home.md?plain=1#L43-L55 specifically is something that you should not do if you can possibly avoid it. GCP should be authorized via SSO (for a developer) or IAM or workload identity or identity federation (for a production system, or for CI).

Reply:
I tried setting `GOOGLE_CREDENTIALS` to `~/.config/gcloud/application_default_credentials.json` and

```diff
diff --git src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java
index 24e119c..3357142 100644
--- src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java
+++ src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java
@@ -202,7 +202,6 @@ class ITUtil {
         SecretBytes bytes = SecretBytes.fromBytes(CREDENTIALS.getBytes(StandardCharsets.UTF_8));
         JsonServiceAccountConfig sac = new JsonServiceAccountConfig();
         sac.setSecretJsonKey(bytes);
-        assertNotNull(sac.getAccountId());
         Credentials credentials = new GoogleRobotPrivateKeyCredentials(PROJECT_ID, sac, null);
 
         CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins);
```

but just get a failure later

```
com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials$AccountIdNotSetException
	at com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials.getGoogleCredential(GoogleRobotPrivateKeyCredentials.java:229)
	at com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials.getGoogleCredential(GoogleRobotPrivateKeyCredentials.java:45)
	at com.google.jenkins.plugins.computeengine.client.ClientUtil.getGoogleCredential(ClientUtil.java:104)
	at com.google.jenkins.plugins.computeengine.client.ClientUtil.getClientFactory(ClientUtil.java:53)
	at com.google.jenkins.plugins.computeengine.client.ClientUtil.getClientFactory(ClientUtil.java:70)
	at com.google.jenkins.plugins.computeengine.integration.ITUtil.initClient(ITUtil.java:254)
	at com.google.jenkins.plugins.computeengine.integration.ComputeEngineCloudOneShotInstanceIT.init(ComputeEngineCloudOneShotInstanceIT.java:84)
```

It seems like there is a lot of code baked into this plugin and its credentials dep which makes unwarranted assumptions about the sort of authn to use to GCP.